homeassistant/helpers/network.py

```diff
@@ -37,6 +39,12 @@ def get_url(
     prefer_cloud: bool = False,
 ) -> str:
     """Get a URL to this instance."""
+    # If the request URL is required, used that
+    if require_current_request:
+        return _get_current_request_url(
```

Comment:
Given this implementation, we should maybe just make it a new method ?

Reply:
🤷  I found it kinda pretty there is just a single point to get this information from.

Both will do.

Reply:
Thinking, I could use the existing methods for this actually, that will make all other parameters valid at that point as well.

Reply:
I've simplified and integrated it fully into all current logic in c3b304d.
The advantage (besides removing a lot of code), is that all current features of `get_url` will work as well.